apps/web/src/pages/templates/components/templates-store/TemplatesStoreModal.tsx

```diff
@@ -0,0 +1,177 @@
+      },
+      {
+        name: ':fa-solid fa-gift: Refferal link',
+        description:
+          'Lorem ipsum dolor sit amet consectetur adipisicing elit. Cupiditate quas totam quod beatae. Ipsam quasi fugiat commodi adipisci eligendi necessitatibus cumque aliquam, dicta natus cupiditate suscipit voluptatum rerum debitis. Ipsum!',
+      },
+    ],
+  },
+  {
+    name: 'Authentification',
+    templates: [
+      {
+        name: ':fa-solid fa-wand-magic-sparkles: Magic link',
+        description:
+          'Lorem ipsum dolor sit amet consectetur adipisicing elit. Cupiditate quas totam quod beatae. Ipsam quasi fugiat commodi adipisci eligendi necessitatibus cumque aliquam, dicta natus cupiditate suscipit voluptatum rerum debitis. Ipsum!',
+      },
+      {
+        name: ':fa-solid fa-unlock: Password change',
+        description:
+          'Lorem ipsum dolor sit amet consectetur adipisicing elit. Cupiditate quas totam quod beatae. Ipsam quasi fugiat commodi adipisci eligendi necessitatibus cumque aliquam, dicta natus cupiditate suscipit voluptatum rerum debitis. Ipsum!',
+      },
+      {
+        name: ':fa-solid fa-unlock: Password change2',
+        description:
+          'Lorem ipsum dolor sit amet consectetur adipisicing elit. Cupiditate quas totam quod beatae. Ipsam quasi fugiat commodi adipisci eligendi necessitatibus cumque aliquam, dicta natus cupiditate suscipit voluptatum rerum debitis. Ipsum!',
+      },
+      {
+        name: ':fa-solid fa-unlock: Password change3',
+        description:
+          'Lorem ipsum dolor sit amet consectetur adipisicing elit. Cupiditate quas totam quod beatae. Ipsam quasi fugiat commodi adipisci eligendi necessitatibus cumque aliquam, dicta natus cupiditate suscipit voluptatum rerum debitis. Ipsum!',
+      },
+    ],
+  },
+];
+
+const getTemplateDetails = (templateName: string): { name: string; iconName: IconName } => {
+  const regexResult = /^:.{1,}: /.exec(templateName);
+  let name = '';
+  let iconName = 'fa-solid fa-question';
+  if (regexResult !== null) {
+    name = templateName.replace(regexResult[0], '').trim();
+    iconName = regexResult[0].replace(/:/g, '').trim();
+  }
+
+  return { name, iconName: iconName as IconName };
+};
+
+export interface ITemplatesStoreModalProps {
+  isOpened: boolean;
+  onClose: () => void;
+}
+
+export const TemplatesStoreModal = ({ isOpened, onClose }: ITemplatesStoreModalProps) => {
+  const theme = useMantineTheme();
+  const { classes: modalClasses } = useStyles();
+  const [selectedTemplate, setTemplate] = useState(TEMPLATES_GROUPED[0].templates[0]);
+  const { name: selectedTemplateName, iconName: selectedTemplateIconName } = getTemplateDetails(selectedTemplate.name);
+
+  return (
+    <Modal
+      opened={isOpened}
+      withinPortal
+      overlayColor={theme.colorScheme === 'dark' ? colors.BGDark : colors.BGLight}
+      overlayOpacity={0.7}
+      classNames={modalClasses}
+      closeOnClickOutside={false}
```

Comment:
Should not we support closeOnClickOutside?

Reply:
I'll ask Nikolay about this